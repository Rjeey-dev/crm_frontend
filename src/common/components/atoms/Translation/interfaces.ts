import {LocalizeContextProps, TranslateProps} from "react-localize-redux";

export interface IProps extends LocalizeContextProps {
    source: TranslateProps,
}