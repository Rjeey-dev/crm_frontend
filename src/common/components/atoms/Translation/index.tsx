import React from 'react';
import {InitializeOptions, Translate, TranslatePlaceholderData, withLocalize} from "react-localize-redux";

import {IProps} from "./interfaces";

function Translation(props: IProps) {
    const data = [] as unknown as TranslatePlaceholderData;

    if (props.source.data) {
        const source = props.source.data;

        Object.keys(props.source.data).map((element: any) => {

            const item = source[element];

            data[element] = <b className='highlighted'>{item}</b>;
        });
    }

    return <Translate id={props.source.id} data={data} options={{ renderInnerHtml: true } as InitializeOptions}/>;
}

export default withLocalize(Translation);

