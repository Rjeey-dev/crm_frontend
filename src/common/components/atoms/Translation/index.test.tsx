import { shallow } from 'enzyme';
import React from 'react';

import Translation from './index';

it("Renders <Translation/> with correct properties", () => {
    const output = shallow(
        <Translation source={{id: 'some_id'}}/>
    ).html();

    expect(output).toMatchSnapshot();
});

it("Renders <Translation/> with data", () => {
    const output = shallow(
        <Translation source={{id: 'some_id_with_dynamic_translation', data: {name: 'name'}}}/>
    ).html();

    expect(output).toMatchSnapshot();
});
