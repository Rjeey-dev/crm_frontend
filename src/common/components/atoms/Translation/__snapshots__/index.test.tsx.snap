// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders <Translation/> with correct properties 1`] = `"Missing translationId: some_id for language: \${ languageCode }"`;

exports[`Renders <Translation/> with data 1`] = `"Missing translationId: some_id_with_dynamic_translation for language: \${ languageCode }"`;
